src/core/thread/meshcop_dataset.cpp

```diff
@@ -239,6 +253,26 @@ ThreadError Dataset::Set(const otOperationalDataset &aDataset, bool aActive)
         Set(tlv);
     }
 
+    if (aDataset.mIsChannelMaskPage0Set)
+    {
+        MeshCoP::ChannelMaskTlv tlv;
+        MeshCoP::ChannelMaskEntry *entry;
+        uint8_t data[6];
+        tlv.Init();
+        tlv.SetLength(sizeof(MeshCoP::ChannelMaskEntry) + sizeof(aDataset.mChannelMaskPage0));
+
+        entry = reinterpret_cast<MeshCoP::ChannelMaskEntry *>(data);
+        entry->SetChannelPage(0);
+        entry->SetMaskLength(sizeof(aDataset.mChannelMaskPage0));
+
+        data[2] = (aDataset.mChannelMaskPage0 >> 24) & 0xff;
+        data[3] = (aDataset.mChannelMaskPage0 >> 16) & 0xff;
+        data[4] = (aDataset.mChannelMaskPage0 >> 8) & 0xff;
+        data[5] = (aDataset.mChannelMaskPage0) & 0xff;
```

Comment:
The channel mask is encoded in little-endian (least-significant bit of the first byte is channel 0).